src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -376,9 +385,18 @@ bool ColorControlServer::computeNewColor16uValue(ColorControlServer::Color16uTra
     }
     else
     {
-        newValue32u = ((uint32_t)(p->initialValue - p->finalValue));
-        newValue32u *= ((uint32_t)(p->stepsRemaining));
-        newValue32u /= ((uint32_t)(p->stepsTotal));
+        newValue32u = static_cast<uint32_t>(p->initialValue - p->finalValue);
+        newValue32u *= static_cast<uint32_t>(p->stepsRemaining);
+
+        /*
+            stepsTotal should always be at least 1,
+            still, prevent division by 0 and skips a meaningless division by 1
+        */
+        if (p->stepsTotal > 1)
```

Comment:
stepsTotal should always be at least 1, should we change to ` if (p->stepsTotal >= 1)`